src/lib/utils/SynapseClient.ts

```diff
@@ -2594,44 +2610,59 @@ export const getAvailableFilesToDownload = (
       console.error('Error on getDownloadListV2 ', err)
       throw err
     })
+
+}
+
+/**
+ * Clear all files from the user's Download List v2.
+ * http://rest-docs.synapse.org/rest/DELETE/download/list.html
+ */
+export const clearDownloadListV2 =
+  (accessToken: string | undefined):Promise<void> => {
+  return doDelete(
+    '/repo/v1/download/list',
+    accessToken,
+    undefined,
+    BackendDestinationEnum.REPO_ENDPOINT,
+  )
+}
+
+/**
+ * Get Download List v2 available files to download
+ * http://rest-docs.synapse.org/rest/POST/download/list/query/async/start.html
+ */
+export const getAvailableFilesToDownload = (
+  request: AvailableFilesRequest,
+  accessToken: string | undefined = undefined,
+): Promise<DownloadListQueryResponse> => {
+  return getDownloadListJobResponse(accessToken, request)
```

Comment:
Would it be possible to specify which QueryResponseDetails object (maybe with a generic) we get in each SynapseClient method instead of casting when the function is called?

Reply:
If I understand what you're looking for correctly, you'd like it to be easier for the caller (so they don't need to get cast the QueryResponseDetails).  I'll update the code and you can let me know if you like it...